### DISCUSSION TITLE
Client side password policies validation improvements

### DISCUSSION THREAD

**Discussion by PotatoesMaster** [keycloak/keycloak#34553] [Ideas]
Client-side validation of password policies is already in place during user registration (#32846).

I wanted to extend this to check for the regex pattern policy. I started working on it (https://github.com/PotatoesMaster/keycloak/commit/c4d16115333a00ed4aed697c84e2a423a0febaf0), but now I think client side validation could be more complete.

~In addition to user registration, client-side validation could be enabled during password update (in [`login-update-password.ftl`](https://github.com/keycloak/keycloak/blob/main/themes/src/main/resources/theme/keycloak.v2/login/login-update-password.ftl)).  
Maybe a macro could be declared to prevent duplicating the password validation bits in `login-update-password.ftl` and `register.ftl`.~ **Edit**: already done in #35479

Client-side check for the password confirmation could be added too.

What are your thoughts about all of this ?

**Comment by StefanLobbenmeierObjego** [comment]
Looks like it was extracted into a macro in https://github.com/keycloak/keycloak/pull/35479:
https://github.com/keycloak/keycloak/blob/0e4b629d915480a638843ab1009caf90b23b1711/themes/src/main/resources/theme/keycloak.v2/login/register.ftl#L65-L66

**Reply by PotatoesMaster** [reply]
Oh! Nice, I didn't see that. Thanks!
